docs/src/main/tut/typeclasses.md

```diff
@@ -0,0 +1,73 @@
+The typeclass pattern is a technique borrowed from haskell. You can
+think of it as an "interface" in the java sense, but universally
+quantified over some type. In scala they are encoded as a trait that
+has one type variable, such as:
```

Comment:
Spire also has used (and probably will continue to use) multi-parameter type classes.